xarray/core/rolling.py

```diff
@@ -785,11 +785,14 @@ def construct(
             if not keep_attrs:
                 dataset[key].attrs = {}
 
+        # Need to stride coords as well. TODO: is there a better way?
+        coords = self.obj.isel(
```

Comment:
```suggestion
        coords = self.obj.coords.to_dataset().isel(
```

Perhaps this is slightly better?

Reply:
I've tried running asv locally based on your suggestion. I've try increased `nx` and `nt` to 3000 and 10000 respectively to see if there's a gap.  I think I don't see much difference here.
New:
```
[33.33%] ··· rolling.DatasetRollingMemory.peakmem_1drolling_construct                                                                                                                                                                                      ok
[33.33%] ··· ======== ======
              stride
             -------- ------
               None    577M
                5      583M
                50     578M
             ======== ======

[66.67%] ··· rolling.DatasetRollingMemory.peakmem_1drolling_reduce                                                                                                                                                                                         ok
[66.67%] ··· ====== ====== =======
             --     use_bottleneck
             ------ --------------
              func   True   False
             ====== ====== =======
              sum    581M    1.6G
              max    575M   1.61G
              mean   575M    1.6G
             ====== ====== =======

[100.00%] ··· rolling.DatasetRollingMemory.peakmem_ndrolling_reduce                                                                                                                                                                                         ok
[100.00%] ··· ====== ====== =======
              --     use_bottleneck
              ------ --------------
               func   True   False
              ====== ====== =======
               sum    595M   1.61G
               max    596M   1.62G
               mean   618M   1.61G
              ====== ====== =======
```

Old:
```
[33.33%] ··· rolling.DatasetRollingMemory.peakmem_1drolling_construct                                                                                                                                                                                      ok
[33.33%] ··· ======== ======
              stride
             -------- ------
               None    577M
                5      577M
                50     578M
             ======== ======

[66.67%] ··· rolling.DatasetRollingMemory.peakmem_1drolling_reduce                                                                                                                                                                                         ok
[66.67%] ··· ====== ====== =======
             --     use_bottleneck
             ------ --------------
              func   True   False
             ====== ====== =======
              sum    576M    1.6G
              max    575M    1.6G
              mean   575M    1.6G
             ====== ====== =======

[100.00%] ··· rolling.DatasetRollingMemory.peakmem_ndrolling_reduce                                                                                                                                                                                         ok
[100.00%] ··· ====== ====== =======
              --     use_bottleneck
              ------ --------------
               func   True   False
              ====== ====== =======
               sum    604M   1.62G
               max    601M   1.61G
               mean   618M   1.62G
              ====== ====== =======
```

Reply:
> Perhaps this is slightly better?

Then I get:
> ValueError: Dimensions {'index'} do not exist. Expected one or more of Frozen({})